subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -649,6 +649,8 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 		case 'p':
 			params->psk = state->optarg;
 			params->psk_length = strlen(params->psk);
+			params->sae_password = optarg;
+			params->sae_password_length = strlen(params->sae_password);
```

Comment:
no point in using the both,  as both have different rules, may be we should add a new option for SAE with no length restrictions.

Reply:
Not using both, just set SAE password by '-p'

Reply:
I mean, setting both `psk` and `sae_password`. `-p` for `psk` and `-s` (if not already used) for `sae_password`.

Reply:
Maybe we don't need to set `params->sae_password`?
currently WPA3 SAE connection works fine.

Reply:
Both password for WPA2 and WPA3 are psk, even through psk for WPA2 is derive PTK, for WPA3 is get P value from EC， but for user, there is no difference, so I prepare to delete this change, and remove the 'sae_password' element in 'wifi_connect_req_params' later. Then psk used for both WPA2 and WPA3.
What do you think about this?

Reply:
The usage is same, but the length restrictions are differnet, `psk` -> `8..63` and `sae_password` is `8..` (no limit).